frontend/src/hooks/use-keypair-update.tsx

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useState } from 'react'
+
+import { requestPassphrase } from '../components/passphrase-modal'
+import { AppToaster } from '../components/toaster'
+import { Intent } from '../config/intent'
+import type { GlobalActions } from '../contexts/global/global-actions'
+import type { GlobalDispatch } from '../contexts/global/global-context'
+import { useGlobal } from '../contexts/global/global-context'
+import { createLogger } from '../lib/logging'
+
+const logger = createLogger('Metadata')
+
+export type Meta = {
+  key: string
+  value: string
+}
+
+export const useKeypairUpdate = (
+  dispatch: GlobalDispatch,
+  actions: GlobalActions,
+  pubKey?: string,
+  wallet?: string
+) => {
+  const { service } = useGlobal()
+  const [loading, setLoading] = useState(false)
+
+  const update = useCallback(
+    async (metadata: Meta[]) => {
+      setLoading(true)
+      try {
+        if (!pubKey || !wallet) {
+          return
```

Comment:
yes, they can be typed, empty returns are of type `void`. I've added an explicit type for the function, hope that's less confusing, since i don't want to return anything from here.